source/ldap.d

```diff
@@ -16,6 +16,28 @@ class LDAPException : Exception
 	}
 }
 
+/// If username has a domain specified, this will split it into a username and domain and return them in a string array.
+/// Params:
+///   username: username to split if domain is specified
+/// Returns: string array with username for the first member and domain for the second. Domain is an empty string if not in the username
+private string[] domainSplit(string username)
+{
+    import std.array : split;
+	import std.algorithm.mutation : reverse;
+
+    auto ret = username.split("@");
```

Comment:
I would personally change it to use indexOf as split allocates memory.

The function also still allocate memory on each of the returns. 
Maybe better to add `ref string[2] returnArray`as a argument or just  `ref string firstPart, ref string secondPart`.

Reply:
I would use findSplit instead, indexOf is ugly